text/0000-light-dom.md

```diff
@@ -0,0 +1,315 @@
+  #shadow-root (open)
+  | <div>
+  |   <b>Blue Shadow:</b>
+  |   <span class="counter">...</span>
+  |   <button type="button">Add one</button>
+  | </div>
+</app-counter-blue-shadow>
+```
+
+_Light DOM_
+
+```html
+<app-container-blue-light>
+    <div>
+      <b>Blue Light:</b>
+      <span class="counter">...</span>
+      <button type="button">Add one</button>
+    </div>
+</app-counter-blue-light>
+```
+
+As a result, when the content of a component resides in the Light DOM, it can be accessed like any other content in the Document host, and thus behave like any other content (styling, APIs, accessibility, third party tooling...).
+
+## Motivation
+
+Consumer applications require DOM traversal and observability of an application’s anatomy from the document root. Without this, theming becomes hard and 3rd party applications do not run properly:
+
+- **Theming and branding**
+
+  Because of the CSS isolation, theming is made harder. Typically, theming is done via APIs (CSS properties) and/or CSS theming parts (`::part`). Both come with caveats and issues. Theming is critical for consumer apps, where custom branding is a must have. Some new APIs, like `::theme`, are investigated but they won’t be pervasively available before years.
+  [Styling is critical to web component reuse, but may prove difficult in practice](https://component.kitchen/blog/posts/styling-is-critical-to-web-component-reuse-but-may-prove-difficult-in-practice)
+
+- **Third party integrations**
+
+  Third party tools need to traverse the DOM, which breaks with Shadow DOM and the existing browser APIs (querySelector, ...). Note that the use of Light DOM fixes for Light DOM components, but not for native Shadow DOM ones if the page contains any: Analytics tools, Personalization platforms, Commerce tools like PriceSpider or Honey...
+
+- **Testing software**
+
+  Tools like Selenium, Cypress etc. face the same issues as third party tools when it comes to traversing the DOM.
+
+Introducing components rendering to the Light DOM opens new possibilities where applications can render most of their content in the Light DOM, but also keep individual widgets (e.g. `select` tag) and leaf components that render in the Shadow DOM. This approach combines the best of both worlds by solving the issues presented above and offering strong encapsulation when needed.
+![shadow spectrum](./shadow-spectrum.png?raw=true "Shadow Spectrum")
+
+### Prior Art
+
+Most of the libraries designed to support Shadow DOM also propose a Light DOM option, with a variety of Shadow DOM features (slots, scoped styles, etc.). It includes:
+
+- [StencilJS](https://stenciljs.com/docs/styling#shadow-dom-in-stencil)
+- [LitElement](https://lit-element.polymer-project.org/api/classes/_lit_element_.litelement.html#createrenderroot)
+- [MS Fast Element](https://fast.design/docs/fast-element/working-with-shadow-dom#shadow-dom-configuration)
+
+## Detailed design
+
+### `LightningElement.shadow`
```

Comment:
I don't think it's too late to change this, the API is not considered stable today. We debated between @nolanlawson and @abdulsattar this couple of weeks ago. I purposefully made the API name generic because I had in mind to reuse the same `shadow` static field in the future to developers configure the shadow root:

```js
// default - shadow DOM (mode: open)
class extends LightningElement {} 

// shadow DOM (mode: open)
class extends LightningElement {
  static shadow = true;
}

// light DOM
class extends LightningElement {
  static shadow = false;
}

// shadow DOM (mode: closed)
class extends LightningElement {
  static shadow = {
    mode: 'closed'
}
```

@caridy I know that you tinkered with something similar in the past. Where did your previous research lead you?

Reply:
Another argument in favor of `shadow = false` over `light = true`: web developers tend to think of "shadow DOM" as a fancy new thing, and "light DOM" as the default. In fact, "light DOM" was just "the DOM" for decades.

So developers are more likely to immediately understand `shadow = false` compared to `light = true`, since they understand `shadow` as this weird new bleeding-edge-standards thing that they need to have a separate mental model for, as opposed to `light`, which is just how the web worked for decades.

One counter-argument to that, though, is that developers working with LWC have probably had to acquaint themselves with `shadow` already, as it's the default in LWC. So I see good arguments both ways.